### PR TITLE
Extend ltinyiiod

### PR DESCRIPTION
**Pull request by cristiantpop**
Extend libtinyiiod: 

- Add `tintiiod_do_writebuf()` function, that allows us to write to the device.
- Handle multiple instances, when more then one client is connected to the board, trough network interface.

### CODE REVIEW

File: tinyiiod.c [hunk 1]
```diff
@@ -186,6 +186,33 @@ void tinyiiod_do_close(struct tinyiiod *iiod, const char *device)
...
+		if(ret > 0) {
+			ret = iiod->ops->write_data(device, buf, offset, ret);
+			offset += ret;
+			if (ret < 0)
+				return ret;
+			bytes_count -= ret;
+		} else
+			return ret;
+	}
+	ret = iiod->ops->store_data(device, total_bytes);
```
Comment (mirceacaprioru) [inline tinyiiod.c]: does this store in memory ? and write_value to device ?

File: tinyiiod.c [hunk 2]
```diff
@@ -186,6 +186,16 @@ void tinyiiod_do_close(struct tinyiiod *iiod, const char *device)
 	tinyiiod_write_value(iiod, ret);
 }
 
+int32_t tinyiiod_do_open_instance(struct tinyiiod *iiod)
+{
+	return iiod->ops->open_instance();
```
Comment (mirceacaprioru) [inline tinyiiod.c:191]: Do we need these function wrappers ?

Comment (cristiantpop) [reply tinyiiod.c:191]: Yes and no.
If we want to remove the wrappers, we will have to expose the "tinyiiod" structure into "tinyiiod.h". In this way, the user of libtinyiiod will have access to the elements of the structure.

File: tinyiiod.c (hunk 1 above)
Comment (cristiantpop) [reply tinyiiod.c]: yes

**Comment by cristiantpop** [comment]
V2:

- rename "store_data()" to a more suggestive name: "transfer_data()"; this is the reverse function of "capture_data()"

**Review by mirceacaprioru — APPROVED** [review]

**Comment by dbogdan** [comment]
transfer_data() and capture_data() are still a little bit confusing for me. I'm thinking of searching something more clear (e.g., transmit_data(), receive_data())...

**Comment by cristiantpop** [comment]
V3:

-  rename "capture_data()" to "transfer_dev_to_mem()";
-  rename "transfer_data()" to "transfer_mem_to_dev()";
- check if the user offers implementation for "transfer_mem_to_dev()", before calling it.

**Comment by cristiantpop** [comment]
V4:

- Add space after "while" instance: "while()" becomes "while ()"

File: tinyiiod.c [hunk 3]
```diff
@@ -186,6 +186,34 @@ void tinyiiod_do_close(struct tinyiiod *iiod, const char *device)
...
+			     const char *device, size_t bytes_count)
+{
+	int32_t ret;
+	size_t total_bytes = bytes_count;
+	char buf[256];
+	size_t offset = 0;
+
+	tinyiiod_write_value(iiod, bytes_count);
+	while (bytes_count) {
+		size_t bytes = bytes_count > sizeof(buf) ? sizeof(buf) : bytes_count;
```
Comment (dbogdan) [inline tinyiiod.c]: I would declare all the variables at the top of the function - otherwise, C99 standard is mandatory.

File: tinyiiod.c [hunk 4]
```diff
@@ -186,6 +186,34 @@ void tinyiiod_do_close(struct tinyiiod *iiod, const char *device)
 	tinyiiod_write_value(iiod, ret);
 }
 
+int32_t tinyiiod_do_writebuf(struct tinyiiod *iiod,
+			     const char *device, size_t bytes_count)
+{
+	int32_t ret;
+	size_t total_bytes = bytes_count;
+	char buf[256];
+	size_t offset = 0;
```
Comment (dbogdan) [inline tinyiiod.c]: I would declare the same type variables on one line and I would order the variable declarations from longest to shortest line.

**Comment by cristiantpop** [comment]
V5: 

- declare the same type variables on one line and order the variable declarations from longest, to shortest line
- declare all the variables at the top of the function

**Review by dbogdan — APPROVED** [review]

**Pull request merged**